core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -646,8 +646,17 @@ private[spark] class AppStatusListener(
   }
 
   override def onUnpersistRDD(event: SparkListenerUnpersistRDD): Unit = {
-    liveRDDs.remove(event.rddId)
-    kvstore.delete(classOf[RDDStorageInfoWrapper], event.rddId)
+    while (true) {
```

Comment:
This code may never cause infinite loop.
Since `liveRDDs` is just a `HashMap` (not a `concurrentHashMap` or others), update by other threads may not be visible in this loop.

Reply:
Thank you for letting me know about a single thread. 

I agree with your point. As you pointed out, when `rdd.isEmpty()` continues to return `false`, it will get into an infinite loop. Thus, I imagine that this works under multithread.

Reply:
hi kiszk, thank you for your thread safe remind.